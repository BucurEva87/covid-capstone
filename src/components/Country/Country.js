import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { Link } from 'react-router-dom';
import style from './Country.module.scss';

const Country = ({ name, flag }) => (
  <article>
    <h2>
      <Link to={name}>{name}</Link>
    </h2>
    <img src={flag} alt="Country flag not available" className={style.flag} />
  </article>
);

Country.propTypes = {
  name: PT.string.isRequired,
  flag: PT.string.isRequired,
};

export default Country;
