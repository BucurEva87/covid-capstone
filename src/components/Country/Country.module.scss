.flag {
  width: 20%;
  height: 10rem;
}
