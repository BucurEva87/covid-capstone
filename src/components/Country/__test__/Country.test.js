import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Country from '../Country';

let getAllByRole;

const countries = [
  {
    country: 'Romania',
    confirmed: 2556,
    recovered: 0,
    deaths: 114,
    flag: 'https://cdn.jsdelivr.net/npm/country-flag-emoji-json@2.0.0/dist/images/RO.svg',
  },
  {
    country: 'England',
    confirmed: 13872,
    recovered: 0,
    deaths: 1774,
    flag: 'https://cdn.jsdelivr.net/npm/country-flag-emoji-json@2.0.0/dist/images/EN.svg',
  },
];

beforeEach(() => {
  const component = render(
    <Provider store={store}>
      <Router>
        <Country
          key={countries[0].country}
          name={countries[0].country}
          country={countries[0].country}
          confirmed={countries[0].confirmed}
          recovered={countries[0].recovered}
          deaths={countries[0].deaths}
          flag={countries[0].flag}
        />
        <Country
          key={countries[1].country}
          name={countries[1].country}
          country={countries[1].country}
          confirmed={countries[1].confirmed}
          recovered={countries[1].recovered}
          deaths={countries[1].deaths}
          flag={countries[1].flag}
        />
      </Router>
    </Provider>,
  );

  getAllByRole = component.getAllByRole;
});

describe('Country component', () => {
  test('is rendered two times on the screen', () => {
    expect(getAllByRole('article')).toHaveLength(2);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Country
            key={countries[0].country}
            name={countries[0].country}
            country={countries[0].country}
            confirmed={countries[0].confirmed}
            recovered={countries[0].recovered}
            deaths={countries[0].deaths}
            flag={countries[0].flag}
          />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
