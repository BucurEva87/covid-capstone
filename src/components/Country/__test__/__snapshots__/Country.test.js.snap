// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component matches snapshot 1`] = `
<article>
  <h2>
    <a
      href="/Romania"
      onClick={[Function]}
    >
      Romania
    </a>
  </h2>
  <img
    alt="Country flag not available"
    className="flag"
    src="https://cdn.jsdelivr.net/npm/country-flag-emoji-json@2.0.0/dist/images/RO.svg"
  />
</article>
`;
